qiskit_experiments/framework/base_experiment.py

```diff
@@ -479,9 +478,27 @@ def set_analysis_options(self, **fields):
         )
         self.analysis.options.update_options(**fields)
 
-    def _postprocess_transpiled_circuits(self, circuits: List[QuantumCircuit], **run_options):
-        """Additional post-processing of transpiled circuits before running on backend"""
-        pass
+    def _transpile(
+        self, circuits: List[QuantumCircuit], backend: Backend, **transpile_options
```

Comment:
Probably backend is not necessary since you can access to it via `self.backend` (also this will bypass `_set_backend` hook). Transpile options and circuit too. Likely you want to make it static method -- is there any plan to hack this method? Something like injecting circuit which is not generated by this experiment instance?

Reply:
It is necessary because `self.backend` could be None, in which case the backend is passed to `BaseExperiment.run` and forwarded to this function.

Reply:
It should not be a static method because transpilation can depend on the experiment instance.

Reply:
This is not correct because existence of backend is guaranteed by these lines. Seems like this allows a hack that injecting backend which is not associated with the instance. Currently you are using `self` to only access deprecated method.
https://github.com/Qiskit/qiskit-experiments/blob/317b964b8cbde0febcda8696fd4bd5cce3490e0a/qiskit_experiments/framework/base_experiment.py#L251-L259

Reply:
Perhaps you are assuming the situation the user directly call this method of the instance initialized without backend to check the physical circuit? Then I agree it's worth having backend here. But I think circuit and transpile options can come from self.

Reply:
Still, the backend needs to be forwarded because it uses `experiment.backend` which could be different from `self.backend`.

Reply:
> Perhaps you are assuming the situation the user directly call this method of the instance initialized without backend to check the physical circuit? Then I agree it's worth having backend here. But I think circuit and transpile options can come from self.

Yes, exactly. If the user calls `run` with a backend specified, then using `self.backend` would be incorrect. I actually this this a bit confusing, and that only way of passing the backend should be supported. That would also eliminate the need for the complicated backend selection code. But that is a separate issue.